packages/js/src/utils/compiler-helper-dependency.ts

```diff
@@ -31,21 +32,49 @@ const jsExecutors = {
   } as const,
 } as const;
 
+export interface GetHelperDependencyOptions {
+  /**
+   * @default false
+   */
+  returnDependencyIfFound?: boolean;
+
+  /**
+   * Use the provided project graph instead of pulling a new one.
+   */
+  projectGraph?: ProjectGraph;
+}
+
 /**
  * Check and return a DependencyNode for the compiler's external helpers npm package. Return "null"
  * if it doesn't need it or it cannot be found in the Project Graph
  *
  * @param {HelperDependency} helperDependency
  * @param {string} configPath
  * @param {DependentBuildableProjectNode[]} dependencies
- * @param {boolean=false} returnDependencyIfFound
+ * @param {GetHelperDependencyOptions={returnDependencyIfFound: false}} options
  */
-export function getHelperDependency(
+export async function getHelperDependency(
+  helperDependency: HelperDependency,
+  configPath: string,
+  dependencies: DependentBuildableProjectNode[],
+  options: GetHelperDependencyOptions = { returnDependencyIfFound: false }
+): Promise<DependentBuildableProjectNode | null> {
+  return getHelperDependencySync(
+    helperDependency,
+    configPath,
+    dependencies,
+    options.projectGraph ?? (await createProjectGraphAsync()),
```

Comment:
I noticed that the `createProjectGraphAsync()` ALWAYS writes a `nxdeps.json` cache whenever it is called. So when you have many executors running in parallel (tested in my workspace with 137 projects) there will be teams where one task will be reading and another will be writing at the same time resulting in a read error of something like this.

```
Error: CloseBraceExpected in /home/runner/work/shipyard/shipyard/node_modules/.cache/nx/nxdeps.json at position 2138112
```

So perhaps calling the create graph function isn't the right solution?
Or maybe the issue is that the create graph should not be rewriting to that file so often?

Reply:
This was a bit more of a band-aid fix than I'd have liked anyways, I'm spending some more time on this now.